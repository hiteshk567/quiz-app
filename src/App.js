import React, { Suspense, useEffect } from "react";
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch,
} from "react-router-dom";

import Header from "./components/navigation/Header";
import "./App.css";
import LoadingSpinner from "./components/UIElements/LoadingSpinner";
const CategorySelection = React.lazy(() =>
  import("./containers/CategoryContainer")
);
const QuizContainer = React.lazy(() => import("./containers/QuizContainer"));
const Result = React.lazy(() => import("./containers/ResultContainer"));

function App({ fetchCategories }) {
  useEffect(() => {
    fetchCategories();
  }, []);

  const routes = (
    <Switch>
      <Route path="/" exact>
        <CategorySelection />
      </Route>
      <Route path="/category/:categoryId" exact>
        <QuizContainer />
      </Route>
      <Route path="/result" exact>
        <Result />
      </Route>
      <Redirect to="/" />
    </Switch>
  );

  return (
    <div className="App">
      <Router>
        <Header />
        <main>
          <Suspense
            fallback={
              <div>
                <LoadingSpinner />
              </div>
            }
          >
            {routes}
          </Suspense>
        </main>
      </Router>
    </div>
  );
}

export default App;
