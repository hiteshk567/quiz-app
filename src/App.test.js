import React from "react";
import ReactDOM, { unmountComponentAtNode } from "react-dom";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import reduxThunk from "redux-thunk";

import App from "./containers/AppContainer";
import { applyMiddleware } from "redux";

let mockStore = configureStore([reduxThunk]);
let store;
let container = null;
beforeEach(() => {
  store = mockStore({
    loading: false,
    category: ["Gaming", "Sports", "General Knowledge"],
    quiz: [],
    user: { name: "Manish", score: 50 },
    completedList: null,
  });
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders the app component without crash", () => {
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>,
    container
  );
});

it("matches the snapshot", () => {
  const tree = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>
  );
  expect(tree).toMatchSnapshot();
});
