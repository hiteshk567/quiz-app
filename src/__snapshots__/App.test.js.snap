// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="main-header"
  >
    <h1
      className="main-navigation__title"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        QUIZ APP
      </a>
    </h1>
    <nav
      className="main-navigation__header-nav"
    >
      <ul
        className="nav-links"
      >
        <li>
          <a
            aria-current="page"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            HOME
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <div
      className="card entry-card"
    >
      <form
        onSubmit={[Function]}
      >
        <h1>
          QUIZ FORM
        </h1>
        <hr />
        <div
          className="form-control"
        >
          <label
            htmlFor="name"
          >
            Name
          </label>
          <input
            id="name"
            onChange={[Function]}
            placeholder="Enter your name..."
            type="text"
            value=""
          />
        </div>
        <h4>
          SELECT QUIZ CATEGORY
        </h4>
        <div>
          <select
            onChange={[Function]}
          >
            <option />
            <option />
            <option />
          </select>
        </div>
        
      </form>
    </div>
  </main>
</div>
`;
